### DISCUSSION TITLE
Installation problem

### DISCUSSION THREAD

**Discussion by eddiemorris135** [rpoleski/MulensModel#72] [General]
Hi Radek,

I came across your program on the https://www.microlensing-source.org/software/ website and I was keen to install it as a hobbyist and explore micro lensing as a new topic. However, I had some difficulties installing the program with the following error:

 ERROR: Failed building wheel for MulensModel
Failed to build MulensModel
ERROR: Could not build wheels for MulensModel which use PEP 517 and cannot be installed directly

Perhaps you have a suggestion how I could proceed?
Thanks
Ed

**Comment by rpoleski** [comment]
Can you please copy-paste longer part of output and the command you're running? 

In general, some PIP 517 issues are solved by updating pip or setuptools package.

**Comment by eddiemorris135** [comment]
Hi.

THanks for the fast reply. I have attached the output here. 

[MulensModel_output.txt](https://github.com/rpoleski/MulensModel/files/10429406/MulensModel_output.txt)

**Comment by rpoleski** [comment]
My knowledge about installing python packages on Windows is very limited. 
From what I see, the error is caused by `M_LN2` in `source\AdaptiveContouring\adaptive_contour.c`. The solution for that would be to add:

```c
#define M_LN2 0.69314718055994530942
```

at the top of that file. But this is a hack, not a real solution. Maybe @ketozhang can help?

**Reply by ketozhang** [reply]
Unfortunately installing MulensModel  on Windows is not supported right now. The source code has always been written with Linux in mind and is incompatible with Windows. 

If you're using Windows, I recommend using Windows Subsystem Linux (WSL) to work with all your Python needs.

**Comment by eddiemorris135** [comment]
Ok,

Thanks for letting me know.

**Reply by rpoleski** [reply]
If you find a solution, then please post it here. Maybe it will be useful for others using Windows.

**Comment by eddiemorris135** [comment]
Update for those of you using a Windows operating system, this solution worked for me:

1) Install Oracle VM VirtualBox Manger with Ubuntu as operating system
2) Install Anaconda and run Spyder IDE
3) In the Spyder IDE enter Conda activate and then pip install MulensModel

Currently this setup is running nicely for me.
